cpp/src/arrow/dataset/api.h

```diff
@@ -22,5 +22,6 @@
 #include "arrow/dataset/file_base.h"
 #include "arrow/dataset/file_csv.h"
 #include "arrow/dataset/file_feather.h"
+#include "arrow/dataset/file_parquet.h"
```

Comment:
I'd expect the `#ifdef` to be here since the other file format headers are included here. Why not?

Reply:
parquet is bit different since it is a separate library. The headers referenced in `file_parquet.h` might not be shipped with the installed arrow.

I also think it's fine that dataset would require parquet.

Reply:
In practice, Dataset does require Parquet today since that's the only kind of data file we support. 

I see the following headers in "arrow/dataset/file_parquet.h":

```
#include <memory>
#include <string>

#include "arrow/dataset/file_base.h"
#include "arrow/dataset/type_fwd.h"
#include "arrow/dataset/visibility.h"
```

So what's the problem?

I'm struggling to see how this is my problem (from R) and not a problem that (also?) needs to be resolved inside `cpp/src/arrow/dataset`, but it's easy enough for me to add the include in R, and I can do that if you say "just do that".

Reply:
Leave it there, we'll make dataset depends on parquet.